### PR TITLE
sync test chart & scripts

### PR DESCRIPTION
**Pull request by hbandura**

### CODE REVIEW

File: helm-charts/sync-test-node/scripts/is_synced.sh [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+LATEST=`kubectl -n $NAMESPACE exec $POD -- geth --verbosity 0 attach --exec 'eth.getBlock("latest").timestamp'`
+RESULT=$?
+# If the execution returned 0, and the output is a number...
+if [ $RESULT -ne 0 ]; then
+    echo "kubectl returned error $RESULT"
+    exit 102
+fi 
+MARK=`date +%s`
+if [ $LATEST -eq $LATEST 2> /dev/null ]; then
```
Comment (jcortejoso) [inline helm-charts/sync-test-node/scripts/is_synced.sh:42]: I think `2> /dev/null` is not necessary here.

File: helm-charts/sync-test-node/scripts/is_synced.sh [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+if [ "$#" -ne 2 ]; then
+    echo "Usage: is_synced.sh [namespace] [docker-tag]"
+    exit 1001
+fi
+
+NAMESPACE=$1
+TAG=$2
+
+# Find out what the pod is
+POD=`kubectl -n $NAMESPACE get pods -o name -l app=sync-test-node -l dockertag=$TAG`
```
Comment (jcortejoso) [inline helm-charts/sync-test-node/scripts/is_synced.sh:26]: `...` is usually considered as deprecated syntax, being $(...) preferred ([SC2006](https://github.com/koalaman/shellcheck/wiki/SC2006))

File: helm-charts/sync-test-node/scripts/is_synced.sh [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+LATEST=`kubectl -n $NAMESPACE exec $POD -- geth --verbosity 0 attach --exec 'eth.getBlock("latest").timestamp'`
+RESULT=$?
+# If the execution returned 0, and the output is a number...
+if [ $RESULT -ne 0 ]; then
+    echo "kubectl returned error $RESULT"
+    exit 102
+fi 
+MARK=`date +%s`
+if [ $LATEST -eq $LATEST 2> /dev/null ]; then
+    DIFF="$(($LATEST - $MARK))"
```
Comment (jcortejoso) [inline helm-charts/sync-test-node/scripts/is_synced.sh:43]: I would use `DIFF="$((LATEST - MARK))"` to use the same syntax as in the next line

File: helm-charts/sync-test-node/scripts/is_synced.sh [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        echo "Sync successful"
+        exit 0
+    else
+        echo "Sync not completed. Latest block is $((-DIFF)) seconds old"
+        exit 2
+    fi
+else
+    echo "eth.getBlock('latest').timestamp returned a non-number"
+    exit 11
+fi
```
Comment (jcortejoso) [inline helm-charts/sync-test-node/scripts/is_synced.sh:55]: You can consider leaving a new line at EOF ([reason](https://thoughtbot.com/blog/no-newline-at-end-of-file))

File: helm-charts/sync-test-node/scripts/is_synced.sh [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+# 102 - kubectl command returned an error when executing the geth attach command
+#
+# 1001 - Invalid command usage
+
+# We consider a good sync if the latest block synced is OLDEST_ACCEPTABLE seconds old
+OLDEST_ACCEPTABLE=30
+
+if [ "$#" -ne 2 ]; then
+    echo "Usage: is_synced.sh [namespace] [docker-tag]"
+    exit 1001
```
Comment (jcortejoso) [inline helm-charts/sync-test-node/scripts/is_synced.sh:19]: Allowed exit codes go from 0 to 255

File: helm-charts/sync-test-node/scripts/is_synced.sh [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+fi 
+MARK=`date +%s`
+if [ $LATEST -eq $LATEST 2> /dev/null ]; then
+    DIFF="$(($LATEST - $MARK))"
+    SUM="$((DIFF + OLDEST_ACCEPTABLE))"
+    if [ "$SUM" -gt 0 ]; then
+        echo "Sync successful"
+        exit 0
+    else
+        echo "Sync not completed. Latest block is $((-DIFF)) seconds old"
```
Comment (jcortejoso) [inline helm-charts/sync-test-node/scripts/is_synced.sh:49]: Is there any difference to `-$DIFF` syntax?

File: helm-charts/sync-test-node/scripts/wait_for_sync.sh [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+TAG=$2
+TIMEOUT=$3
+
+if [ "$TIMEOUT" -lt "$SLEEP_SECONDS" ]; then
+  SLEEP_SECONDS=$TIMEOUT
+fi
+
+END_TIME=$((SECONDS+TIMEOUT))
+
+while [ $SECONDS -lt $END_TIME ]; do
```
Comment (jcortejoso) [inline helm-charts/sync-test-node/scripts/wait_for_sync.sh:26]: $SECONDS in not available in some sh shells ([SC3028](https://github.com/koalaman/shellcheck/wiki/SC3028)). You can 'force' the shebang to bash.

File: helm-charts/sync-test-node/scripts/wait_for_sync.sh [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+# 0 - sync successful
+# 1 - timeout reached
+# 11 - error testing for sync
+
+# Time between sync polls
+SLEEP_SECONDS=60
+
+if [ "$#" -ne 3 ]; then
+    echo "Usage: wait_for_sync.sh [namespace] [docker-tag] [timeout_in_seconds]"
+    exit 1001
```
Comment (jcortejoso) [inline helm-charts/sync-test-node/scripts/wait_for_sync.sh:13]: Exit code between 0 and 255

File: helm-charts/sync-test-node/scripts/stop.sh [hunk 9]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+
+if [ "$#" -ne 2 ]; then
+    echo "Usage: stop.sh [namespace] [docker-tag]"
+    exit 1001
```
Comment (jcortejoso) [inline helm-charts/sync-test-node/scripts/stop.sh:5]: Exit code between 0 and 255

File: helm-charts/sync-test-node/scripts/start.sh [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+    exit 1001
+fi
+
+NAMESPACE=$1
+TAG=$2
+SYNCMODE=$3
+RELEASE="sync-test-$TAG"
+
+helm install ./sync-test-node --namespace $NAMESPACE \
+ --name $RELEASE --set dockertag=$TAG --set geth.syncmode=$SYNCMODE
```
Comment (jcortejoso) [inline helm-charts/sync-test-node/scripts/start.sh:14]: Newline at EOF

File: helm-charts/sync-test-node/scripts/stop.sh [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+fi
+
+NAMESPACE=$1
+TAG=$2
+RELEASE="sync-test-$TAG"
+
+helm del --purge $RELEASE
```
Comment (jcortejoso) [inline helm-charts/sync-test-node/scripts/stop.sh:12]: Newline at EOF

File: helm-charts/sync-test-node/scripts/wait_for_sync.sh [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+        # Syncing, not done
+        :
+    else
+        echo "is_synced.sh returned $RESULT, terminating"
+        exit 11
+    fi
+done
+
+echo "Timeout reached"
+exit 1
```
Comment (jcortejoso) [inline helm-charts/sync-test-node/scripts/wait_for_sync.sh:44]: Newline at EOF

File: helm-charts/sync-test-node/values.yaml [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+dockertag: 13a3406014e7df867b8d7305a0adef51b5d48211 
+
+geth:
+  image:
+    repository: us.gcr.io/celo-testnet/geth
+    imagePullPolicy: IfNotPresent
+  resources:
+    requests:
+      memory: "2048Mi"
+      cpu: "2000m"
```
Comment (jcortejoso) [inline helm-charts/sync-test-node/values.yaml:15]: Seems a bit high to the values I am used to use, but I am ok with being generous. Also consider `requests` is a soft limit, so the pod could use more resources if the node where it runs have free resources.

**Review by jcortejoso — CHANGES_REQUESTED** [review]
Nice PR. I am leaving some comments based most on https://www.shellcheck.net/.

File: helm-charts/sync-test-node/scripts/is_synced.sh (hunk 6 above)
Comment (hbandura) [reply helm-charts/sync-test-node/scripts/is_synced.sh:49]: If DIFF is -3, I want it to show '3' (diff is negative usually)

Comment (hbandura) [reply helm-charts/sync-test-node/scripts/is_synced.sh:49]: will -$DIFF show '3' or '--3' ?

File: helm-charts/sync-test-node/scripts/is_synced.sh (hunk 1 above)
Comment (hbandura) [reply helm-charts/sync-test-node/scripts/is_synced.sh:42]: it prints 'sh: [: : integer expression expected' if i remove it when LATEST is not a number

Comment (jcortejoso) [reply helm-charts/sync-test-node/scripts/is_synced.sh:42]: I think that is bc you need to quotes with `"` the variables, so `[ "$LATEST" -eq "$LATEST" ]`. Another check we could use instead is `[ -z "$LATEST" ]` that checks if $LATEST is defined and not empty string.

File: helm-charts/sync-test-node/scripts/is_synced.sh (hunk 6 above)
Comment (jcortejoso) [reply helm-charts/sync-test-node/scripts/is_synced.sh:49]: Oh, I was not considering that $DIFF could be negative. Then is perfect as it is.

**Review by jcortejoso — APPROVED** [review]
lgtm

**Pull request merged**